### ISSUE TITLE
Regression test on reshare protocol

### ISSUE THREAD

**Issue by JesseAbram** [entropyxyz/entropy-core#1210]
Currently we test the reshare by mocking it. We test each peace individually but there is a timing aspect to the communication between the chain and tss that is just mocked. This is because triggering a new session (for testing) is hard to do in a substrate chain 

more thought needs to go into how to test this 

some ideas 

1) a new chain runtime with a signigicantly shorter session time 
2) chopstick (however issues with sending http calls in chopsticks)

**Comment by ameba23** [comment]
Good idea, i think this would greatly improve our tests for anything related to what goes on in the session handler. 

Can we set the session length in the chainspec?  Or would we need to compile the chain node with a special `short-session` flag or something.
